### ISSUE TITLE
Point type fails in mutation response and query

### ISSUE THREAD

**Issue by keanugrieves** [neo4j/graphql#278]
**Describe the bug**
I've had the same issue as #220 beginning with 1.0.2 and I just assumed it would magically fix itself when I got closer to production and was ready to deploy a new DB, but this time, I was able to reproduce it with a fresh DB instance and a fresh install of the latest neo4j releases. It creates the node just fine, but it returns a strange error message anytime you query for a Point.

**Type definitions**
```
type User {
  id: ID! @readonly
  meta_enabled: Boolean @default(value: true)
  meta_createdAt: DateTime! @timestamp(operations: [CREATE])
  meta_updatedAt: DateTime! @timestamp(operations: [CREATE, UPDATE])
  meta_robotsUpdatedAt: DateTime @readonly
  auth_email: String! @readonly
  auth_lastLoginFrom: Point!
  auth_password: String! @private
  auth_role: UserAuthRole! @readonly
  name_first: String!
  name_last: String!
}
```

**Mutation:**
```
mutation {
  createUsers(
    input: [
      {
        id: "1234567890123456"
        auth_email: "user@test.com"
        auth_lastLoginFrom: {
          longitude: -123.1381874
          latitude: 44.0341663
        }
        auth_role: USER
        name_first: "Test"
        name_last: "User"
      }
    ]
  ) {
    users {
      id
      auth_lastLoginFrom {
        latitude
        longitude
      }
    }
  }
}
```

**Response:**
```
{
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Point.latitude.",
      "locations": [
        {
          "line": 8,
          "column": 9
        }
      ],
      "path": [
        "createUsers",
        "users",
        0,
        "auth_lastLoginFrom",
        "latitude"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot return null for non-nullable field Point.latitude.",
            "    at completeValue (...\node_modules\\graphql\\execution\\execute.js:559:13)",
            "    at resolveField (...\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (...\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (...\node_modules\\graphql\\execution\\execute.js:748:10)",
            "    at completeObjectValue (...\node_modules\\graphql\\execution\\execute.js:738:10)",
            "    at completeValue (...\node_modules\\graphql\\execution\\execute.js:590:12)",
            "    at completeValue (...\node_modules\\graphql\\execution\\execute.js:556:21)",
            "    at resolveField (...\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (...\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (...\node_modules\\graphql\\execution\\execute.js:748:10)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

**Log:**
```
  @neo4j/graphql:graphql Incoming GraphQL:
Query:
mutation {
  createUsers(
    input: [{id: "1234567890123456", auth_email: "user@test.com", auth_lastLoginFrom: {longitude: -123.1381874, latitude: 44.0341663}, auth_role: USER, name_first: "Test", name_last: "User"}]
  ) {
    users {
      id
      auth_lastLoginFrom {
        latitude
        longitude
      }
    }
  }
}
Variables:
{} +7m
  @neo4j/graphql:auth Verifying JWT +7m
  @neo4j/graphql:execute About to execute Cypher:
Cypher:
CALL {
CREATE (this0:User)
SET this0.meta_createdAt = datetime()
SET this0.meta_updatedAt = datetime()
SET this0.id = $this0_id
SET this0.meta_enabled = $this0_meta_enabled
SET this0.auth_email = $this0_auth_email
SET this0.name_first = $this0_name_first
SET this0.name_last = $this0_name_last
SET this0.auth_role = $this0_auth_role
SET this0.auth_lastLoginFrom = point($this0_auth_lastLoginFrom)
RETURN this0
}

RETURN
this0 { .id, auth_lastLoginFrom: { point: this0.auth_lastLoginFrom } } AS this0
Params:
{
  "this0_id": "1234567890123456",
  "this0_meta_enabled": true,
  "this0_auth_email": "user@test.com",
  "this0_name_first": "Test",
  "this0_name_last": "User",
  "this0_auth_role": "USER",
  "this0_auth_lastLoginFrom": {
    "longitude": -123.1381874,
    "latitude": 44.0341663
  }
} +7m
  @neo4j/graphql:execute Execute successful, received 1 records +140ms
```

**To Reproduce**
See above.

**Expected behavior**
That it would return the Point object.

**Screenshots**

**System (please complete the following information):**

-   OS: [Windows]
-   @neo4j/graphql version: 1.03
-   @neo4j-graphql-ogm version: 1.03
-   neo4j Enterprise Edition version: 4.3
-   apollo-server-express version: 2.25.1
-   Node.js version: 14.17.1

**Additional context**
This also occurs when I just query for `users` including `auth_lastLoginFrom`, so let me know if you'd like to see that response and log.

**Comment by keanugrieves** [comment]
When I tried this last time, I was using @neo4j/graphql 1.02, @neo4j/graphql-ogm 1.02, a local install of neo4j Enterprise Edition 4.2 and Node.js version 12.x. This time, I'm using a remotely hosted neo4j EE database.

I've also tried using the new alpha releases of @neo4j/graphql and @neo4j/graphql-ogm, if that helps.

**Comment by maurom88** [comment]
I encountered the same issue with the Point type. It's worth noting, though, that even if the mutation returns an error, the node is actually created in the database.

I hope the additional context below can help you figure it out.

Type definition:

```
type Geolocation {
  uuid: ID! @id
  location: Point!
}
```

Query:

```
query getLocations {
  geolocations {
    uuid
    location{
        latitude
        longitude
      }
  }
}
```

Response:

```
{
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Point.latitude.",
      "locations": [
        {
          "line": 5,
          "column": 7
        }
      ],
      "path": [
        "geolocations",
        0,
        "location",
        "latitude"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot return null for non-nullable field Point.latitude.",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:559:13)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)",
            "    at completeObjectValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:738:10)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:590:12)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:556:21)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)"
          ]
        }
      }
    },
    {
      "message": "Cannot return null for non-nullable field Point.latitude.",
      "locations": [
        {
          "line": 5,
          "column": 7
        }
      ],
      "path": [
        "geolocations",
        1,
        "location",
        "latitude"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot return null for non-nullable field Point.latitude.",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:559:13)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)",
            "    at completeObjectValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:738:10)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:590:12)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:556:21)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)"
          ]
        }
      }
    },
    {
      "message": "Cannot return null for non-nullable field Point.latitude.",
      "locations": [
        {
          "line": 5,
          "column": 7
        }
      ],
      "path": [
        "geolocations",
        2,
        "location",
        "latitude"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot return null for non-nullable field Point.latitude.",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:559:13)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)",
            "    at completeObjectValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:738:10)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:590:12)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:556:21)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)"
          ]
        }
      }
    }
  ],
  "data": {
    "geolocations": [
      null,
      null,
      null
    ]
  }
}
```

Mutation:

```
mutation createlocations($longitude: Float!, $latitude: Float!) {
 createGeolocations(input: [{location: {longitude: $longitude, latitude: $latitude}}]) {
   geolocations {
     location {
       longitude
       latitude
     }
   }
 } 
}

Variables:
{
  "longitude": 10,
  "latitude": 10
}
```

Response:

```
{
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Point.longitude.",
      "locations": [
        {
          "line": 7,
          "column": 9
        }
      ],
      "path": [
        "createGeolocations",
        "geolocations",
        0,
        "location",
        "longitude"
      ],
      "extensions": {
        "code": "INTERNAL_SERVER_ERROR",
        "exception": {
          "stacktrace": [
            "Error: Cannot return null for non-nullable field Point.longitude.",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:559:13)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)",
            "    at completeObjectValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:738:10)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:590:12)",
            "    at completeValue (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:556:21)",
            "    at resolveField (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:472:19)",
            "    at executeFields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:292:18)",
            "    at collectAndExecuteSubfields (C:\[redacted]\\node_modules\\graphql\\execution\\execute.js:748:10)"
          ]
        }
      }
    }
  ],
  "data": null
}
```

Node created in the database despite the error:
![nodes in neo4j db](https://user-images.githubusercontent.com/21005055/123522036-e49ed980-d677-11eb-8e50-7f0c9710107c.png)


System:

```
OS: [Windows]
@neo4j/graphql version: 1.0.0
@neo4j-graphql-ogm version: 1.0.0
neo4j Desktop Edition version: 1.4.5
apollo-server version: 2.24.1
apollo-server-express version: 2.25.0
Node.js version: 14.17.0
```

**Comment by darrellwarde** [comment]
@keanugrieves and @maurom88, thank you so much for the detailed examples, but I can't for the life of me reproduce them, they both work perfectly on my end!

Out of interest, are you able to run the Cypher provided in your debug output directly against your database, and let me know what the result is there? Intrigued to see if this is happening at the database or GraphQL level.

**Comment by keanugrieves** [comment]
@darrellwarde 

**Input (Round 1):**

```
CALL {
CREATE (this0:User)
SET this0.meta_createdAt = datetime()
SET this0.meta_updatedAt = datetime()
SET this0.id = "1234567890123456"
SET this0.meta_enabled = true
SET this0.auth_email = "user@test.com"
SET this0.name_first = "Test"
SET this0.name_last = "User"
SET this0.auth_role = "USER"
SET this0.auth_lastLoginFrom = point({
    "longitude": -123.1381874,
    "latitude": 44.0341663
})
RETURN this0
}

RETURN
this0 { .id, auth_lastLoginFrom: { point: this0.auth_lastLoginFrom } } AS this0
```

**Output (Round 1):**

```
Invalid input '{': expected "+" or "-" (line 11, column 38 (offset: 341))
"SET this0.auth_lastLoginFrom = point({"
```

**Input (Round 2):**

```
CALL {
CREATE (this0:User)
SET this0.meta_createdAt = datetime()
SET this0.meta_updatedAt = datetime()
SET this0.id = "1234567890123456"
SET this0.meta_enabled = true
SET this0.auth_email = "user@test.com"
SET this0.name_first = "Test"
SET this0.name_last = "User"
SET this0.auth_role = "USER"
SET this0.auth_lastLoginFrom = point({
    longitude: -123.1381874,
    latitude: 44.0341663
})
RETURN this0
}

RETURN
this0 { .id, auth_lastLoginFrom: { point: this0.auth_lastLoginFrom } } AS this0
```

**Output (Round 2):**

```
{
  "auth_lastLoginFrom": {
"point": point({srid:4326, x:-123.1381874, y:44.0341663})
  },
  "id": "1234567890123456"
}
```

Is it because the GraphQL input function is stringifying latitude and longitude?

I currently have my server code torn apart, so I can't test the query. 😩

**Comment by darrellwarde** [comment]
> Is it because the GraphQL input function is stringifying latitude and longitude?

By stringifying, do you mean the "latitude" and "longitude" keys in the object? If so, no, that's just a difference to how it's represented in parameters vs directly in the query. If you had used Neo4j Browser and set the parameters using:

```
:params 
  "this0_id": "1234567890123456",
  "this0_meta_enabled": true,
  "this0_auth_email": "user@test.com",
  "this0_name_first": "Test",
  "this0_name_last": "User",
  "this0_auth_role": "USER",
  "this0_auth_lastLoginFrom": {
    "longitude": -123.1381874,
    "latitude": 44.0341663
  }
```

And then executed the Cypher using:

```
CALL {
CREATE (this0:User)
SET this0.meta_createdAt = datetime()
SET this0.meta_updatedAt = datetime()
SET this0.id = $this0_id
SET this0.meta_enabled = $this0_meta_enabled
SET this0.auth_email = $this0_auth_email
SET this0.name_first = $this0_name_first
SET this0.name_last = $this0_name_last
SET this0.auth_role = $this0_auth_role
SET this0.auth_lastLoginFrom = point($this0_auth_lastLoginFrom)
RETURN this0
}

RETURN
this0 { .id, auth_lastLoginFrom: { point: this0.auth_lastLoginFrom } } AS this0
```

Then that would have worked as expected.

The output from your second example is exactly what I would expect to see - `x` and `y` being returned in the point, which subsequently get translated to `longitude` and `latitude` in field resolvers for the `Point` type, so this is still a bit of a dead end! 🙁

**Comment by keanugrieves** [comment]
Ah well, here's hoping someone comes along with a more enlightening example. For now, I'll just store the long and lat as separate properties so I can still query by distance. Thanks!

**Comment by darrellwarde** [comment]
This _should_ now be fixed by #297! 🤞  Our apologies for the delay in finding this one, it was a difficult bug to track down but then very obvious once we've found it! We're going to get it merged and released ASAP.
